### PR TITLE
API-7065

### PR DESCRIPTION
**Pull request by peteslater-ee**

### CODE REVIEW

File: app/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationController.scala [hunk 1]
```diff
@@ -115,6 +126,54 @@ class OpenApiDocumentationController @Inject() (
       }
   }
 
+  def fetchOasResolved(service: String, version: String) = Action.async { implicit request =>
+    def handleSuccess(openApi: OpenAPI): Result = Ok(Yaml.pretty.writeValueAsString(openApi)).withHeaders(HeaderNames.CONTENT_DISPOSITION -> "attachment; filename=\"application.yaml\"")
+    val handleFailure: Result = NotFound
+
+    val parseOptions  = new ParseOptions();
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationController.scala:133]: Do we need the `;`?

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -19,7 +21,20 @@ object AppDependencies {
...
+    "com.typesafe.play"                     %% "play-json-joda"               % playJsonVersion,
+    "io.swagger.parser.v3"                  %  "swagger-parser"               % "2.1.9"
+      excludeAll(
+        ExclusionRule("com.fasterxml.jackson.core", "jackson-databind"),
+        ExclusionRule("com.fasterxml.jackson.core", "jackson-core"),
+        ExclusionRule("com.fasterxml.jackson.core", "jackson-annotations"),
+        ExclusionRule("com.fasterxml.jackson.dataformat", "jackson-dataformat-yaml"),
+        ExclusionRule("com.fasterxml.jackson.datatype", "jackson-datatype-jsr310")
+      ),
+    "com.fasterxml.jackson.core" % "jackson-core" % jacksonVersion,
```
Comment (johnsgp) [inline project/AppDependencies.scala:33]: Would  be nice to format these in columns, like the lines above.

File: test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      loggedInUserService,
+      errorHandler,
+      navigationService,
+      openAPIV3ParserMock
+    )
+  }
+
+  "OpenApiDocumentationController" should {
+    "successfully fetch resolved OAS specification" in new Setup {
+      when(appConfig.oasFetchResolvedMaxDuration).thenReturn(20000)
```
Comment (johnsgp) [inline test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala:75]: Could we reduce the timeout for the tests?

File: test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      when(openAPIV3ParserMock.readLocation(*, *, *)).thenReturn(openApiSwaggerParseResult)
```
Comment (johnsgp) [inline test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala:76]: Is it worth doing a MockModule for this mock?

File: app/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationController.scala (hunk 1 above)
Comment (peteslater-ee) [reply app/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationController.scala:133]: Ah... noooo

File: test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala (hunk 4 above)
Comment (peteslater-ee) [reply test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala:76]: Prolly not - it's only for this test

File: test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala (hunk 3 above)
Comment (peteslater-ee) [reply test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala:75]: Maybe... I'll try

File: project/AppDependencies.scala (hunk 2 above)
Comment (peteslater-ee) [reply project/AppDependencies.scala:33]: I'll format the whole codebase again

File: test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala (hunk 3 above)
Comment (peteslater-ee) [reply test/uk/gov/hmrc/apidocumentation/controllers/OpenApiDocumentationControllerSpec.scala:75]: I changed it to 1000 which should never get hit as the test returns from the mock instantly

**Review by johnsgp — APPROVED** [review]

**Pull request merged**